SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationFragment.java

```diff
@@ -219,17 +225,23 @@ private void renderControls(Activity activity) {
     }
 
     private void renderContent(Activity activity) {
-        for (ConversationAtom content : page.getContent()) {
+        for (int i = 0; i < page.getContent().size(); i++) {
+            ConversationAtom content = page.getContent().get(i);
             ConversationStyle conversationStyle = content.getStyle();
             ConversationColorStyle colorStyle = conversationStyle.getBg();
 
             if (content instanceof Content) {
                 Content modelContent = (Content) content;
                 String modelType = modelContent.getType().toString();
                 if (modelType.equalsIgnoreCase(ConversationAtom.TYPE_CONTENT_IMAGE)) {
-                    ConversationImageView iv = new ConversationImageView(activity, modelContent);
                     String filePath = swrveConversation.getCacheDir().getAbsolutePath() + "/" + modelContent.getValue();
                     if(SwrveHelper.hasFileAccess(filePath)) {
+                        ConversationImageView iv;
+                        if (i == 0 && pageBorderRadius > 0) { // only apply rounded corner if its the first atom in page
+                            iv = new ConversationImageViewRounded(activity, modelContent, pageBorderRadius, pageBgColor);
```

Comment:
Yeah, need to clip corners on video and images. Doing it on the whole layout would be better but I couldn't get it to work :(